### DISCUSSION TITLE
I need help understanding this NAME_REGEX for vim.regex

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#27634] [Q&A] [answered]
As title. In some plugin, I found the following Lua string and it's hard for me to understand why some chars need to be escaped: (note that it's going to be consumed by `vim.regex`)

```lua
local NAME_REGEX = '\\%([^/\\\\:\\*?<>\'"`\\|]\\)'
```

Isn't that the `*` doesn't have any special meaning in a character class in vimscript so we don't need to escape it? The same question for the `|` before the close bracket `]`. Thanks for your reading.

I always have a hard time reading regex definition so please help.

**Comment by zeertzjq** [comment, marked as answer]
They don't need to be escaped.

**Reply by nyngwang** [reply]
Thank you!
